### PR TITLE
New opcode cleo return with

### PR DESCRIPTION
**Pull request by MiranDMC**
Implemented cleo_return_with opcode according to https://github.com/sannybuilder/dev/issues/263#issuecomment-1722089145

### CODE REVIEW

File: source/CCustomOpcodeSystem.cpp [hunk 1]
```diff
@@ -3220,6 +3222,47 @@ namespace CLEO
...
+
+		if (returnParamCount) GetScriptParams(cs, returnParamCount);
+
+		ScmFunction* scmFunc = ScmFunction::Store[cs->GetScmFunction()];
+		scmFunc->Return(cs); // jump back to cleo_call, right after last input param. Return slot var args starts here
+		if (scmFunc->moduleExportRef != nullptr) GetInstance().ModuleSystem.ReleaseModuleRef((char*)scmFunc->moduleExportRef); // exiting export - release module
+		delete scmFunc;
+
+		DWORD returnSlotCount = GetVarArgCount(cs);
+		if(returnSlotCount > returnParamCount)
```
Comment (x87) [inline source/CCustomOpcodeSystem.cpp]:
```suggestion
		if(returnSlotCount != returnParamCount)
```

Comment (MiranDMC) [reply source/CCustomOpcodeSystem.cpp]: updated

**Review by x87 — APPROVED** [review]

File: source/CCustomOpcodeSystem.cpp [hunk 2]
```diff
@@ -3220,6 +3222,51 @@ namespace CLEO
...
+		{
+			SHOW_ERROR("Opcode [2002] returned %d params, while function caller expected %d in script %s\nScript suspended.", returnParamCount, returnSlotCount,  cs->GetInfoStr().c_str());
+			return CCustomOpcodeSystem::ErrorSuspendScript(cs);
+		}
+		else if(returnParamCount > 1 && returnParamCount > returnSlotCount) // more args than needed, not critical
+		{
+			LOG_WARNING(thread, "Opcode [2002] returned %d params, while function caller expected %d in script %s", returnParamCount, returnSlotCount, cs->GetInfoStr().c_str());
+		}
+
+		if (returnSlotCount) SetScriptParams(cs, returnSlotCount);
```
Comment (x87) [inline source/CCustomOpcodeSystem.cpp:3264]:
```suggestion
	DWORD returnSlotCount = GetVarArgCount(cs);	
	if(returnParamCount)
	{ 	
		if (returnParamCount != returnSlotCount)
        	{
        		SHOW_ERROR("Opcode [2002] returned %d params, while function caller expected %d in script %s\nScript suspended.", returnParamCount, returnSlotCount,  cs->GetInfoStr().c_str());
        		return CCustomOpcodeSystem::ErrorSuspendScript(cs);
        	}
        	SetScriptParams(cs, returnSlotCount);	
        }
```

**Review by x87 — APPROVED** [review]

**Comment by x87** [comment]
Thanks!

**Pull request merged**